### ISSUE TITLE
Lake toml format disallows `-` in package name

### ISSUE THREAD

**Issue by JamesGallicchio** [leanprover/lean4#4292]
### Prerequisites

Please put an X between the brackets as you perform the following steps:

* [x] Check that your issue is not already filed:
      https://github.com/leanprover/lean4/issues
* [x] Reduce the issue to a minimal, self-contained, reproducible test case.
      Avoid dependencies to Mathlib or Batteries.
* [x] Test your test case against the latest nightly release, for example on
      https://live.lean-lang.org/#project=lean-nightly
      (You can also use the settings there to switch to “Lean nightly”)

### Description

When using toml format, cannot require packages with `-` in the package name.

### Context

Trying to update a project that requires docgen4 to 4.8.0-rc2

### Steps to Reproduce

1. Make an empty lake project with the following `lakefile.toml`:
```
name = "mwe"

[[require]]
name = "doc-gen4"
git  = "https://github.com/leanprover/doc-gen4.git"
rev  = "main"
```
2. Run `lake update`

**Expected behavior:** Correctly parses toml and downloads dependency

**Actual behavior:** Reports an error:
```
error: lakefile.toml:10:7: key name: expected name
```

### Versions

`Lake version 5.0.0-873ef2d (Lean version 4.8.0-rc2)`

### Impact

Add :+1: to [issues you consider important](https://github.com/leanprover/lean4/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc). If others are impacted by this issue, please ask them to add :+1: to it.

**Comment by tydeu** [comment]
@JamesGallicchio While this a problem, there is a workaround. Lake is parsing the string as a Lean `Name`, so you can enclose it in French quotes (e.g., `‹doc-gen4›`) to successfully name the package. (For instance, this is what `lake translate-config` will do.) I do still plan to fix this in a future release.

**Comment by tydeu** [comment]
This has been fixed in #4496 (I forgot this problem had an issue when I did that PR).
